File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/regressiontests/paths/Employee.java
package com.tickaroo.tikxml.regressiontests.paths;

import com.tickaroo.tikxml.annotation.PropertyElement;
import com.tickaroo.tikxml.annotation.Xml;

@Xml
public class Employee extends Person {
  @PropertyElement String name;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Employee)) return false;
    if (!super.equals(o)) return false;

    Employee employee = (Employee) o;

    return name != null ? name.equals(employee.name) : employee.name == null;
  }

  @Override public int hashCode() {
    int result = super.hashCode();
    result = 31 * result + (name != null ? name.hashCode() : 0);
    return result;
  }
}

File: annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/regressiontests/paths/Employee.kt
package com.tickaroo.tikxml.regressiontests.paths

import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml

@Xml
class Employee : Person() {
    @PropertyElement
    var name: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Employee) return false
        if (!super.equals(other)) return false

        val employee = other as Employee?

        return if (name != null) name == employee!!.name else employee!!.name == null
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + if (name != null) name!!.hashCode() else 0
        return result
    }
}